pkg/client/client.go

```diff
@@ -997,8 +997,9 @@ func (c *Client) WaitForDaemonSetRollout(ctx context.Context, ds *appsv1.DaemonS
 				}
 			}
 		}
-
-		if d.Status.NumberAvailable != nodeReadyCount {
+		// In BZ-2004051 it has been noticed that the number of available pods can be greater than
+		// the node ready count.
```

Comment:
nit:
Can we add the reason here in the comment instead of pointing to a BZ?

Reply:
Done